fiat-roles/src/main/java/com/netflix/spinnaker/fiat/providers/DefaultApplicationProvider.java

```diff
@@ -58,6 +62,15 @@ public DefaultApplicationProvider(Front50Service front50Service, ClouddriverServ
           .filter(app -> !appByName.containsKey(app.getName()))
           .forEach(app -> appByName.put(app.getName(), app));
 
+      if (allowAccessToUnknownApplications) {
```

Comment:
idea is that if we're allowing access to unknown applications (aka any that don't explicitly have permissions specified), than we don't need to explicitly include them in the user permissions.

this significantly reduces the size of a `UserPermission` when you're dealing with many thousands of applications of which only a few actually have permissions.